test-data/unit/check-enum.test

```diff
@@ -74,11 +77,11 @@ class N(IntEnum):
     x = 1
     y = 1
 n = 1
-n = N.x  # Subclass of int, so it's okay
+if int():
+    n = N.x  # Subclass of int, so it's okay
```

Comment:
This one is also not clear. I assume we want to test that there is no error even with the new semantics.